server/src/main/java/io/crate/planner/optimizer/LoadedRules.java

```diff
@@ -32,71 +32,21 @@
-import io.crate.planner.optimizer.rule.MergeFilterAndCollect;
-import io.crate.planner.optimizer.rule.MergeFilters;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathFetchOrEval;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathGroupBy;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathJoin;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathOrder;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathProjectSet;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathRename;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathUnion;
-import io.crate.planner.optimizer.rule.MoveFilterBeneathWindowAgg;
-import io.crate.planner.optimizer.rule.MoveOrderBeneathFetchOrEval;
-import io.crate.planner.optimizer.rule.MoveOrderBeneathNestedLoop;
-import io.crate.planner.optimizer.rule.MoveOrderBeneathRename;
-import io.crate.planner.optimizer.rule.MoveOrderBeneathUnion;
-import io.crate.planner.optimizer.rule.OptimizeCollectWhereClauseAccess;
-import io.crate.planner.optimizer.rule.RewriteNestedLoopJoinToHashJoin;
-import io.crate.planner.optimizer.rule.RemoveRedundantFetchOrEval;
-import io.crate.planner.optimizer.rule.RewriteFilterOnOuterJoinToInnerJoin;
-import io.crate.planner.optimizer.rule.RewriteGroupByKeysLimitToLimitDistinct;
-import io.crate.planner.optimizer.rule.RewriteToQueryThenFetch;
+import io.crate.planner.operators.LogicalPlanner;
 import io.crate.types.DataTypes;
 
 @Singleton
 public class LoadedRules implements SessionSettingProvider {
 
     private static final String OPTIMIZER_SETTING_PREFIX = "optimizer_";
 
-    private final List<Class<? extends Rule<?>>> rules = List.of(
-        RemoveRedundantFetchOrEval.class,
-        MergeAggregateAndCollectToCount.class,
-        MergeAggregateRenameAndCollectToCount.class,
-        MergeFilters.class,
-        MoveFilterBeneathRename.class,
-        MoveFilterBeneathFetchOrEval.class,
-        MoveFilterBeneathOrder.class,
-        MoveFilterBeneathProjectSet.class,
-        MoveFilterBeneathJoin.class,
-        MoveFilterBeneathUnion.class,
-        MoveFilterBeneathGroupBy.class,
-        MoveFilterBeneathWindowAgg.class,
-        MergeFilterAndCollect.class,
-        RewriteFilterOnOuterJoinToInnerJoin.class,
-        MoveOrderBeneathUnion.class,
-        MoveOrderBeneathNestedLoop.class,
-        MoveOrderBeneathFetchOrEval.class,
-        MoveOrderBeneathRename.class,
-        DeduplicateOrder.class,
-        OptimizeCollectWhereClauseAccess.class,
-        RewriteGroupByKeysLimitToLimitDistinct.class,
-        RewriteInsertFromSubQueryToInsertFromValues.class,
-        RewriteToQueryThenFetch.class,
-        RewriteNestedLoopJoinToHashJoin.class
-    );
-
     @Override
     public List<SessionSetting<?>> sessionSettings() {
-        return Lists2.map(rules, this::buildRuleSessionSetting);
+        return Lists2.map(LogicalPlanner.optimizerRules(), x -> buildRuleSessionSetting(x.getClass()));
```

Comment:
When declaring a static list inside this class as already suggested:
```
    private static final List<Rule<?>> RULES = Lists2.concat(
            Lists2.concat(
                    ITERATIVE_OPTIMIZER_RULES,
                    FETCH_OPTIMIZER_RULES
            ),
            WRITE_OPTIMIZER_RULES
    );
```

and the suggestion for the `buildRuleSessionSetting`, this could stay similar:
```
        return Lists2.map(RULES, this::buildRuleSessionSetting);
```